README.md

```diff
@@ -1,8 +1,134 @@
+configuration. Flaky tests could be harmful for developers because their
+failures do not always indicate bugs in the code.
+
+This plugin is designed to handle flaky tests, including re-running
+failed tests, aggregate and report flaky tests statistics and so on.
+
+**Note: Currently most features of this plugin are designed for Git,
+Maven and freestyle build projects.**
+
+## Install
+
+1. Get Jenkins.
+2. Install the Cucumber Reports plugin.
+3. Restart Jenkins.
+
+## Features
+
+### Publish re-run information for flaky tests
+
+The plugin includes support for the "rerunFailingTestsCount" option of
+the Maven surefire plug-in which produces additional data about test
+flakiness. It lets users to choose to re-run failed tests up to N times,
+and if it passes within any of those N times, the re-run will stop and
+the test will be marked as a "Flake". The build will be marked as
+successful if there is no failed tests but only "flaky" tests.
+
+### "Deflake" Action for Builds with Failed Tests
+
+The plugin provides a "Deflake" action. It is different from rebuilding the project, as it will:
+1) (Optionally) Checkout the exact revision of that failing build
+2) Only run all previously failed tests
+
+![Deflake Action](images/deflake_label.png)
+
+After clicking "Deflake", there is another configuration window.
+
+![Deflake Action Configuration](images/deflake_config.png)
+
+* **rerunFailingTestsCount**: Sets the number of automatic reruns of failed tests.
+  However if this parameter is already set in the project pom.xml, then it
+  cannot be overwritten.
+
+* **only run failing tests**: when checked, the plugin will generate `-Dtest=...` to
+  pass to Maven, so only previous failed tests will be re-run.
+
+Deflake build will show up as **"Deflake Build \# ..."**.
+
+![Deflake Build information](images/deflake_build.png)
+
+### Statistics in Test Results
+
+Display all the re-runs in the test result page:
+
+![Passed Test Result](images/flaky_pass.png)
+
+Display flaky tests information in the test result table as a badge:
+
+![Test Result Badge](images/flaky_badge.png)
+
+### Statistics over several builds
+
+For each Git revision and each test, we count how many times it passed,
+how many times it failed. So if a test passed and failed for the both
+revision, we count it as a flake; if it always failed, then it's a fail,
+otherwise it's a pass.
+
+If Git is not used as scm, then we use build number as revision number.
+
+![Aggregated Flake History](images/flake_history_aggregated.png)
+
+![Single Flake History](images/flake_history_single.png)
+
+## Usage
+
+### Pipeline project type
+
+The plugin can be used in pipelines together with the [junit-realtime-test-reporter](https://www.jenkins.io/doc/pipeline/steps/junit-realtime-test-reporter/) plugin:
+
+    realtimeJUnit(keepLongStdio: true, testDataPublishers: [[$class: 'JUnitFlakyTestDataPublisher']], testResults: 'target/surefire-reports/TEST*.xml') {
+        sh 'mvn ...'
+    }
```

Comment:
why are we mentioning that other plugin here? We can keep it, but we should first document "vanilla" usage with the standard junit reporter